### DISCUSSION TITLE
Hourly isDayTime Property Incorrect After Last Update?

### DISCUSSION THREAD

**Discussion by adam-russell** [weather-gov/api#735] [General]
I couldn't find any other discussion about this, and not sure if I'm missing anything, but it looks like the isDayTime property on hourly forecast periods is now incorrect at the edges of the day after the update this week.

E.g. from https://api.weather.gov/gridpoints/TBW/74,61/forecast/hourly

Current response snippet:

```
{
    "number": 8,
    "name": "",
    "startTime": "2024-06-22T17:00:00-04:00",
    "endTime": "2024-06-22T18:00:00-04:00",
    "isDaytime": true,
    "temperature": 86,
    "temperatureUnit": "F",
    "temperatureTrend": "",
    "probabilityOfPrecipitation": {
        "unitCode": "wmoUnit:percent",
        "value": 50
    },
    "dewpoint": {
        "unitCode": "wmoUnit:degC",
        "value": 26.111111111111111
    },
    "relativeHumidity": {
        "unitCode": "wmoUnit:percent",
        "value": 80
    },
    "windSpeed": "9 mph",
    "windDirection": "SW",
    "icon": "/icons/land/day/tsra_hi,50?size=small",
    "shortForecast": "Chance Showers And Thunderstorms",
    "detailedForecast": ""
},
{
    "number": 9,
    "name": "",
    "startTime": "2024-06-22T18:00:00-04:00",
    "endTime": "2024-06-22T19:00:00-04:00",
    "isDaytime": false,
    "temperature": 84,
    "temperatureUnit": "F",
    "temperatureTrend": "",
    "probabilityOfPrecipitation": {
        "unitCode": "wmoUnit:percent",
        "value": 60
    },
    "dewpoint": {
        "unitCode": "wmoUnit:degC",
        "value": 26.111111111111111
    },
    "relativeHumidity": {
        "unitCode": "wmoUnit:percent",
        "value": 84
    },
    "windSpeed": "8 mph",
    "windDirection": "SSW",
    "icon": "/icons/land/night/tsra_hi,60?size=small",
    "shortForecast": "Showers And Thunderstorms Likely",
    "detailedForecast": ""
},
```

The sunset is around 8:27PM at this location, so I would think period 9 should still show isDayTime true here. I *believe* this worked fine prior to the update. (The several periods near the edges of days are showing incorrectly for me right now.) Maybe I've misinterpreted or missed something, though? Thanks anyone for any help.

**Comment by JJKraw** [comment]
Right now I'm seeing 0600 - 1800 showing as true and the others as false and I think that's what I've always seen. I've noticed before that it doesn't seem to adjust to the actual rise/set times.

**Reply by adam-russell** [reply]
Thanks @JJKraw, appreciate the details. It does look like I imagined it accounting for rise/set times previously, and it never has.
